### PR TITLE
Imp generar facturae

### PR DESCRIPTION
**Pull request by bapons**
Actualizar contants

### CODE REVIEW

**Comment by bapons** [comment]
Add:

- test
- Tox config
- pyproject.toml
- Coverage
- Github workflow

and more ...

File: tests/test_validations.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from facturae.exceptions import AccountantValidation
+from tests import get_xml_obj__parser
+
```
Comment (marija-milicevic) [inline tests/test_validations.py]: En las pruebas de integracion deberiamos testear solo funcionalidad principal, no cubrimos caminos falsos y 'corner cases'. Esto testeamos con pruebas unitarias. En este caso concreto yo haria una/dos pruebas de integracion para validacion de factura (validate_invoice). Y luego hacer pruebas unitarrias de cada funcionalidad (validate_invoice_tax, validate_invoice_gross_amount, validate_invoice_tax_output...) cubriento todo los casos y escenarios...

Comment (bapons) [reply tests/test_validations.py]: @marija-milicevic se pasan algunos test a test unitarios utilizando mock & parametrize. La parte negativa es que el test ahora queda más acoplado a la lógica de la función.

Comment (marija-milicevic) [reply tests/test_validations.py]: @bapons test unitarios deberian estar acoplado a la logica de la funcion, ya que sirven para verificar que un trozo de codigo funciona bien, independiente del resto de codigo

File: facturae/accountant_validation.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
+from decimal import Decimal
+
+from facturae.constants import TaxTypeCode
+from facturae.exceptions import AccountantValidation
+
+
+class BaseValidation:
+    def _validate_tax_amount(self, tax):
```
Comment (marija-milicevic) [inline facturae/accountant_validation.py]: Para validacion de cuotas hay que tener en cuenta si precision es en nivel de linea o de factura, dependiente de esto en algunos casos se pueden obtener los diferentes resultados - https://txerpa.atlassian.net/wiki/spaces/TXER/pages/38637225/Calculo+de+totales+de+factura

File: tests/test_unit_validations.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+@pytest.mark.parametrize(
+    "sum_response, total_tax_outputs, expect",
+    [
+        ("17.88", "17.88", does_not_raise()),
+        ("17.85", "17.86", pytest.raises(AccountantValidation)),
+        ("17.88", "17.87", pytest.raises(AccountantValidation)),
+    ],
+)
+def test_unit_validate_invoice_tax_output(sum_response, total_tax_outputs, expect):
```
Comment (marija-milicevic) [inline tests/test_unit_validations.py:103]: Que pasa con el caso cuando se rondea el valor?  Por ejemplo , tener sum_response 17.8822 y total_tax_outputs 17.88. En este caso no deberia saltar error porque los totales de factura deberian ir con dos decimales (al menos en caso de EUR).

Comment (bapons) [reply tests/test_unit_validations.py:94]: Modificación. La suma de decimales de impuestos redondeados al impuesto que tiene más decimales. 
Con esto total_tax_outputs debería tener los mismos decimales.

Comment (bapons) [reply tests/test_unit_validations.py:103]: Se asegura _sum_tax_amount con el redondeo de la suma de los importes de impuestos con el impuesto que contenga más decimales. 
Con esto el valor del campo debería coincidir (TotalTaxes...)

File: facturae/accountant_validation.py (hunk 2 above)
Comment (bapons) [reply facturae/accountant_validation.py]: Código obsoleto, se ha eliminado.
